src/sage/combinat/integer_vectors_mod_permgroup.py

```diff
@@ -763,18 +788,192 @@ def __iter__(self):
             [3, 1, 1, 2]
             [3, 0, 2, 2]
             [2, 2, 2, 1]
+
+            Check that :issue:`36681` is fixed::
+            sage: I = IntegerVectorsModPermutationGroup(
+            ....:     PermutationGroup([], domain=[]), sum=0)
+            sage: list(iter(I))     # Single empty vector
+            [[]]
+
+            Check that :issue:`36681` is fixed::
+            sage: I = IntegerVectorsModPermutationGroup(
+            ....:     PermutationGroup([], domain=[]), sum=3)
+            sage: list(iter(I))     # No solutions
+            []
+
+
         """
+        # Special cases when domain is empty.
+        if self.n == 0:
+            if self._sum is not None and self._sum > 0:
+                # No empty vector can have positive sum.
+                return iter(())
+            else:
+                # Sum is allowed to be zero.  It does not matter what
+                # the maxpart is, the empty vector is a solution.
+                return iter([[]])
+
+        # General case, nonempty domain.
         if self._max_part < 0:
             return self.elements_of_depth_iterator(self._sum)
         else:
-            SF = RecursivelyEnumeratedSet_forest((self([0]*(self.n), check=False),),
-                              lambda x : [self(y, check=False) for y in canonical_children(self._sgs, x, self._max_part)],
-                              algorithm='breadth')
+            SF = RecursivelyEnumeratedSet_forest(
+                (self([0]*(self.n), check=False),),
+                lambda x: [self(y, check=False)
+                           for y in canonical_children(
+                                   self._sgs, x, self._max_part)],
+                algorithm='breadth')
             if self._sum is None:
                 return iter(SF)
             else:
                 return SF.elements_of_depth_iterator(self._sum)
 
+    def cardinality(self):
+        r"""
+        Return the number of integer vectors in the set.
+
+        The number is computed using the cycle index theorem, which is
+        faster than listing the vectors.
+
+        EXAMPLES::
+
+            With a trivial group all vectors are canonical::
```

Comment:
As above, the indentation needs to be fixed.  Also, in this case, `EXAMPLES::` should be `EXAMPLES:` (because there will be no indent after).  [this is sphinx, not me :-)]